src/netCDF_extract_utils.R

```diff
@@ -0,0 +1,360 @@
+  # add depth variable
+  dsg$depth <- filter(dsg$sn, .data$value == "depth")
+  # Add profile id variable since featureType = timeSeriesProfile
+  dsg$profile_id <- get_profile_id(nc_meta$attribute)
+  
+  nc_list <- get_nc_list(nc, dsg, nc_meta, read_data)
+  
+  return(add_globals(nc_list, nc_meta))
+}
+
+#' @title pull data for sites
+#' @description for the specified `sites`, pull data for the specified
+#' variable (`var`). The data will be pulled for all dates (for `var`
+#' = 'temp' | 'ice') and for all depths (if `var` is 'temp')
+#' @param nc_file the filepath for the netCDF
+#' @param nc_info the information about the open netCDF, pulled using
+#' `read_timeseries_profile_dsg()`
+#' @param var the variable for which data should be pulled. Must be
+#' either 'ice' or 'temp'
+#' @param sites a vector of site ids for which data will be pulled
+#' @param long_format boolean - should the returned data be in long format
+#' or not. Default = FALSE
+#' @returns a dataframe in long or wide format containing the data for
+#' the specified variable for the specified sites
+pull_data_for_sites <- function(nc_file, nc_info, var, sites, long_format = FALSE) {
+  # Open netCDF
+  nc <-  open.nc(nc_file)
+  on.exit(close.nc(nc), add  = TRUE)
+  
+  # Parameter checks
+  if (!(var %in% c('ice', 'temp'))) {
+    stop(sprintf("The provided var argument (%s) is not a valid variable choice. Valid options are 'ice' or 'temp'", var))
+  } 
+
+  sites_not_in_data <- sites[!(sites %in% nc_info$timeseries_id)]
+  if (length(sites_not_in_data) > 0) {
+    stop(sprintf("Data for site(s) %s are not included in the specified netCDF file", paste(sites_not_in_data, collapse=', ')))
+  }
+  
+  # If possible, group sites based on indices
+  sites <- sites[order(match(sites, nc_info$timeseries_id))] # sort sites based on order in netCDF
+  site_indices <- purrr::map(sites, ~ which(nc_info$timeseries_id == .x)) %>% unlist()
+  index_groups <- cumsum(c(1L, diff(site_indices) != 1))
+  site_index_groups <- split(site_indices, index_groups)
+  
+  # Pull data for requested variable for requested sites
+  data_subset <- purrr::map_dfc(site_index_groups, function(index_group) {
+    # Find sites associated w/ set of indices
+    sites <- nc_info$timeseries_id[index_group]
+    var_data <- as.data.frame(var.get.nc(nc, var,
+                                         start =c(1, index_group[1], 1),
+                                         count = c(length(nc_info$time), length(index_group), length(nc_info$depth))))
+    
+    if (var == 'temp') {
+      colnames(var_data) <- paste0(rep(sites, length(nc_info$depth)),'_', rep(nc_info$depth, each=length(sites)))
+      
+      # Remove excess depth columns (> lake depth), where all temp values = NA
+      var_data <- select(var_data, where(~sum(!is.na(.x)) > 0))
+    } else {
+      colnames(var_data) <- as.character(sites)
+    }
+    return(var_data)
+  }) %>%
+    mutate(time = nc_info$time, .before = 1)
```

Comment:
The `time` column comes through as a `POSIXct` class by default here. I think we should convert to a `Date` class to avoid confusion.

```suggestion
    mutate(time = as.Date(nc_info$time), .before = 1)
```

Reply:
good thinking!